### DISCUSSION TITLE
Is there any way to not load TLS credential on Server?

### DISCUSSION THREAD

**Discussion by Ang0426** [microsoft/msquic#3509] [Q&A]
I do not want to use TLS, and how to keep the QUIC Server up and running ?

I had set Server's and Client's  CredConfig.Type is QUIC_CREDENTIAL_TYPE_NONE  and CredConfig.Flags is QUIC_CREDENTIAL_FLAG_NO_CERTIFICATE_VALIDATION.

But when I start a new connection,  it will callback the event: Shut down by transport

**Comment by anrossi** [comment]
QUIC uses TLS mandatorily, so the server cannot do the handshake without a certificate. For testing, you could create a certificate with OpenSSL and use that every time.

**Reply by Ang0426** [reply]
Hi
thanks you, can I use same CertHash and use it every time?

**Reply by anrossi** [reply]
The CertHash is used to identify a certificate in the Windows Certificate Store, so you could create a certificate in the store, and use the CertHash to use that certificate every time on the same device.
Assuming you want your server program to run on multiple machines, then you'd have to create a certificate on each machine and use the CertHash to use that certificate for that machine.
CertHash is only available on Windows devices, and if using OpenSSL on Windows with CertHash, the certificate and private key must be created as exportable.
